compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -333,15 +333,17 @@ LinalgMatmulOpToLinalgMmt4DOpPattern::chooseTileParams(Value lhs, Value rhs,
                                   ArrayRef<int> m0k0n0ForMatVec,
                                   std::string comment) {
     assert(m0k0n0ForMatVec[2] == 1 && "not a matrix*vector shape");
-    if (shapeN == 1) {
-      return Mmt4DTileParams(m0k0n0ForMatVec, comment + ", matrix*vector");
-    } else if (shapeM == 1) {
+    if (shapeN == 1 || shapeN == 2) {
+      int config[3] = {m0k0n0ForMatVec[0], m0k0n0ForMatVec[1],
+                       static_cast<int>(shapeN)};
+      return Mmt4DTileParams(config, comment + ", matrix*vector");
+    } else if (shapeM == 1 || shapeM == 2) {
       // The vector*matrix case is intentionally derived from the matrix*vector
       // case by swapping M and N dims so that in kernel codegen we can reuse
       // matrix*vector kernels by swapping LHS and RHS.
-      int m0k0n0ForVecMat[3] = {m0k0n0ForMatVec[2], m0k0n0ForMatVec[1],
-                                m0k0n0ForMatVec[0]};
-      return Mmt4DTileParams(m0k0n0ForVecMat, comment + ", vector*matrix");
+      int config[3] = {static_cast<int>(shapeM), m0k0n0ForMatVec[1],
```

Comment:
this is still assuming that the kernel shape for M==2 can be obtained by widening the kernel shape for M==1.   i still think that this function needs instead to take a separate parameter, alongside `m0k0n0` and `m0k0n0ForMatVec`, allowing to design the kernel for the M==2 case independently.

Reply:
Good point. I don't have a good design for this snippet. I added one more function argument in this case, please take a look. Thanks!